sdk/javascript/generator/StructTypeFormatter.py

```diff
@@ -318,7 +333,13 @@ def generate_serialize_field(self, field):
 			serialize_field = field.extensions.printer.store(field_value)
 			serialize_line = f'buffer.write({serialize_field});{field_comment}\n'
 
-		return indent_if_conditional(condition, serialize_line)
+		if not condition:
+			return serialize_line
+
+		serialize_field_default = field.extensions.printer.store(field.extensions.printer.get_default_value())
+		return indent_if_conditional(
+			condition,
+			f'buffer.write({check_unset(f"this.{lang_field_name(field.name)}")} ? {serialize_field_default} : {serialize_field});\n')
```

Comment:
this part is bit weird, you have early quit above, so if we're inside `if condition` it means that most likely `condition` is already checking variable (or well condition) that's about to be serialized...
So I _think_ `check_unset` here is kinda redundant, so I _guess_ you should be able to use same `serialize_line` as above?

Reply:
you're right, can simplify